comms/src/outbound_message_service/outbound_message_pool/outbound_message_worker.rs

```diff
@@ -0,0 +1,311 @@
+            .spawn(move || loop {
+                match worker.run() {
+                    Ok(_) => break Ok(()),
+                    Err(err @ OutboundMessagePoolError::WorkerShutdownSignalDisconnected) => {
+                        error!(target: LOG_TARGET, "Message Pool worker exited: {:?}", err);
+                        error!(
+                            target: LOG_TARGET,
+                            "The shutdown signal disconnected likely because the outbound message pool went out of \
+                             scope before shutdown was called. Exiting worker with and error."
+                        );
+                        break Err(err);
+                    },
+                    Err(err) => {
+                        error!(target: LOG_TARGET, "Outbound Message Pool worker exited: {:?}", err);
+                        warn!(target: LOG_TARGET, "Restarting outbound message worker after failure.");
+                        // Sleep so that if this service continually restarts, we don't get high CPU usage
+                        thread::sleep(Duration::from_secs(1));
+                    },
+                }
+            })
+            .map_err(|_| OutboundMessagePoolError::ThreadInitializationError)?;
+
+        Ok((thread_handle, shutdown_signal))
+    }
+
+    /// Start MessagePoolWorker which will connect to the inbound message dealer, accept messages from the queue,
+    /// attempt to send them and if it cannot send then requeue the message
+    fn run(&mut self) -> Result<(), OutboundMessagePoolError> {
+        // Connection to the message dealer proxy
+        let message_source_connection = Connection::new(&self.context, Direction::Inbound)
+            .set_name("omp-message-source")
+            .set_socket_establishment(SocketEstablishment::Connect)
+            .establish(&self.message_source_address)
+            .map_err(OutboundMessagePoolError::ConnectionError)?;
+
+        let failed_msg_connection = Connection::new(&self.context, Direction::Outbound)
+            .set_name("omp-failed-message")
+            .set_socket_establishment(SocketEstablishment::Connect)
+            .establish(&self.failed_message_queue_address)
+            .map_err(OutboundMessagePoolError::ConnectionError)?;
+
+        loop {
+            match self.shutdown_receiver.recv_timeout(Duration::from_millis(5)) {
+                // Shut down signal received
+                Ok(_) => break,
+                // Nothing received
+                Err(RecvTimeoutError::Timeout) => {},
+                // Sender disconnected before sending the shutdown signal, this is an error
+                Err(RecvTimeoutError::Disconnected) => {
+                    return Err(OutboundMessagePoolError::WorkerShutdownSignalDisconnected)
+                },
+            }
+
+            // Read any new messages to be sent.
+            if let Some(frames) = connection_try!(message_source_connection.receive(1000)) {
```

Comment:
I have this 1000 all over the place, is this the size of the packet.
Should this not rather be put inside a const somewhere in a single place?

Reply:
Nope, it's a timeout value for the connection - a constant could make it clearer but you only need look at the receive function to know what the value means - the value need only be sensible for the loop you're using it in